src/output/output.cpp

```diff
@@ -242,6 +242,28 @@ wayfire_view wf::output_impl_t::get_active_view() const
     return active_view;
 }
 
+bool wf::output_impl_t::can_activate_plugin(const plugin_grab_interface_uptr& owner)
```

Comment:
Since we have all the checks here, maybe we can just reuse this function in `activate_plugin`? (So we don't have duplicate logic)

Reply:
That mostly works. There however there's a check to load the plugin again sandwiched between the initial checks and the loop to check compatibility. If I put the loop in can_activate() then call it with the other initial checks, it becomes before the condition that might load the plugin again. Not sure what to do here exactly.

Reply:
Oh, you should move the sandwiched check to the can_activate_plugin function too. We need to have the same behavior in both functions, otherwise it will just get confusing(also add the ignore_inhibit parameter to can_activate)

Reply:
This sandwiched check potentially adds the plugin and returns early. We don't want to actually add any plugins to the active list in can_activate().

Reply:
Ok, I put all the checks in can_activate except ignore_inhibit and put the sandwiched check again in activate just to output the right message.